src/core/qgsrunprocess.h

```diff
@@ -22,7 +22,10 @@
 #define QGSRUNPROCESS_H
 
 #include <QObject>
+#if QT_CONFIG(process)
```

Comment:
Could you instead move this #if to include the whole QgsBlockingProcess class? Given that it's going to be completely non-functional on these environments, I'd rather skip it and avoid potential confusion...